admiral/pkg/controller/admiral/globaltraffic_test.go

```diff
@@ -25,3 +28,88 @@ func TestNewGlobalTrafficController(t *testing.T) {
 		t.Errorf("GlobalTraffic controller should never be nil without an error thrown")
 	}
 }
+
+func TestGlobalTrafficAddUpdateDelete(t *testing.T) {
```

Comment:
This test isn't asserting anything. Is there a way we can build out a GlobalTrafficHandler that actually has to handle the update? Something like what we have here? https://github.com/istio-ecosystem/admiral/blob/37deea2910e06ea46a32bef2ac308c6c834efd8b/admiral/pkg/clusters/types_test.go#L45

Reply:
I would keep the tests for controller in here, and the tests for the handler in the handler tests. If the controller isn't doing anything other than passing the object to the handler, then the only thing to be tested here is to make sure the handler is being called with the object.

Reply:
That makes sense - my next question: can we assert that? Maybe add a field to the mock handler struct that records params? Because as it stands now, if globalTrafficController.Updated was completely empty, this would pass.

Reply:
Added a simple object tracker in the mock handler.